docs/pme.md

```diff
@@ -0,0 +1,245 @@
+// var pme = require('pme');
+
+[NoInterfaceObject]
+interface PME {
+    begin();
+    forget();
+    configure(unsigned short context,
+              unsigned short classificationMode,
+              unsigned short distanceMode,
+              unsigned short minInfluence,
+              unsigned short maxInfluence);
+    learn(unsigned char[] pattern, unsigned long category);
+    unsigned long classify(unsigned char[] pattern);
+    Neuron readNeuron(unsigned long id);
+    writeVector(unsigned char[] pattern);
+    unsigned short getCommittedCount();
+    unsigned short getGlobalContext();
+    setGlobalContext(unsigned short context);
+    unsigned short getNeuronContext();
+    setNeuronContext(unsigned short context);
+    unsigned short getClassifierMode();
+    setClassifierMode(unsigned short mode);
+    unsigned short getDistanceMode();
+    setDistanceMode(unsigned short mode);
+
+    unsigned short RBF_MODE;        // RBF classification mode
+    unsigned short KNN_MODE;        // KNN classification mode
+    unsigned short L1_DISTANCE;     // L1 distance mode
+    unsigned short LSUP_DISTANCE;   // LSUP distance mode
+    unsigned long NO_MATCH;         // indicate a pattern could not be classified
+    unsigned short MIN_CONTEXT;     // minimum context value
+    unsigned short MAX_CONTEXT;     // maximum context value
+    unsigned long MAX_VECTOR_SIZE;  // Maximum pattern size (in bytes)
+    unsigned long FIRST_NEURON_ID;  // ID of first neuron in network
+    unsigned long LAST_NEURON_ID;   // ID of last neuron in network
+    unsigned long MAX_NEURONS;      // Number of neurons in the network
+};
+
+[NoInterfaceObject]
+interface Neuron {
+    unsigned short category;
+    unsigned short context;
+    unsigned short influence;
+    unsigned short minInfluence;
+};
+```
+
+API Documentation
+-----------------
+### PME.begin
+
+`void begin();`
+
+Initialize the PME so it is ready for operation.
+
+### PME.forget
+
+`void forget();`
+
+Clear any data committed to the network, making the network ready to learn again.
+
+### PME.configure
+
+`configure(unsigned short context,
+           unsigned short classificationMode,
+           unsigned short distanceMode,
+           unsigned short minInfluence,
+           unsigned short maxInfluence);`
+
+Configure the engine with parameters used for training data.
+
+The `context` is valid context value range between 1-127. A context value of 0 enables all neurons, with no regard to their context.
+The `classificationMode` is the classifying function to use. Valid values are:
+
+ - PME.RBF_MODE (default)
+ - PME.KNN_MODE
+
+The `distanceMode` is the distance function to use. Valid values are:
+
+ - PME.LSUP_DISTANCE
+ - PME.L1_DISTANCE
+
+The `minInfluence` is the minimum influence value used on the neuron.
+The `maxInfluence` is the maximum influence value used on the neuron.
+
+### PME.learn
+
+`void learn(unsigned char[] pattern, unsigned long category);`
```

Comment:
You're taking a JS array? So it's really a "number pattern[]" probably? You need to decide if that makes more sense or a Buffer.